app/tr_php/01/task06.php

```diff
@@ -23,10 +28,116 @@
  *  4冊目 -> タイトル:退屈なことはPythonにやらせよう, 値段:2,200円
 */
 
-$input;
-$input_text;
-$result;
+$input = filter_input_array(INPUT_POST, [
+    'mode' => FILTER_DEFAULT,
+    'category' => FILTER_DEFAULT,
+    'min_price' => FILTER_VALIDATE_INT,
+    'max_price' => FILTER_VALIDATE_INT,
+]);
+
+if (!is_array($input)) {
+    $input = []; // Prevent null access warnings
+}
+
+
+if (empty($input['mode']) && !empty($input['category'])) {
+    $input['mode'] = 'category';
+}
+
+
```

Comment:
細かいですが、どの言語でも「2行以上連続した空行は設定しない」ようにしましょう。
```suggestion
}

if (empty($input['mode']) && !empty($input['category'])) {
    $input['mode'] = 'category';
}

```

Reply:
ご指摘、ありがとうございます！
どの言語でも「2行以上連続した空行は設定しない」とのこと、承知しました。今後、気を付けながらコードを書くようにします。